pkg/resource/table/hooks.go

```diff
@@ -140,139 +150,166 @@ func (rm *resourceManager) customUpdateTable(
+func (rm *resourceManager) syncTable(
+	ctx context.Context,
+	r *resource,
+	delta *ackcompare.Delta,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTable")
+	defer exit(err)
+
+	input, err := rm.newUpdateTablePayload(ctx, r, delta)
+	if err != nil {
+		return err
 	}
-	return tags
-}
 
-// svcTagsFromResourceTags transforms a *v1alpha1.Tag array to a *svcsdk.Tag array.
-func sdkTagsFromResourceTags(rTags []*v1alpha1.Tag) []*svcsdk.Tag {
-	tags := make([]*svcsdk.Tag, len(rTags))
-	for i := range rTags {
-		tags[i] = &svcsdk.Tag{
-			Key:   rTags[i].Key,
-			Value: rTags[i].Value,
-		}
+	_, err = rm.sdkapi.UpdateTable(input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateTable", err)
+	if err != nil {
+		return err
 	}
-	return tags
+	return nil
 }
 
-// computeTagsDelta compares two Tag arrays and return three different list
-// containing the added, updated and removed tags.
-// The removed tags only contains the Key of tags
-func computeTagsDelta(
-	a []*v1alpha1.Tag,
-	b []*v1alpha1.Tag,
-) (added, updated []*v1alpha1.Tag, removed []*string) {
-	var visitedIndexes []string
-mainLoop:
-	for _, aElement := range a {
-		visitedIndexes = append(visitedIndexes, *aElement.Key)
-		for _, bElement := range b {
-			if equalStrings(aElement.Key, bElement.Key) {
-				if !equalStrings(aElement.Value, bElement.Value) {
-					updated = append(updated, bElement)
-				}
-				continue mainLoop
+// newUpdateTablePayload constructs the updateTableInput object.
+func (rm *resourceManager) newUpdateTablePayload(
+	ctx context.Context,
+	r *resource,
+	delta *ackcompare.Delta,
+) (*svcsdk.UpdateTableInput, error) {
+	input := &svcsdk.UpdateTableInput{
+		TableName: aws.String(*r.ko.Spec.TableName),
+	}
+
+	if delta.DifferentAt("Spec.BillingMode") {
+		if r.ko.Spec.BillingMode != nil {
+			input.BillingMode = aws.String(*r.ko.Spec.BillingMode)
+		} else {
+			// set biling mode to the default value `PROVISIONED`
+			input.BillingMode = aws.String(svcsdk.BillingModeProvisioned)
+		}
+	}
+	if delta.DifferentAt("Spec.StreamSpecification") {
+		if r.ko.Spec.StreamSpecification.StreamEnabled != nil {
+			input.StreamSpecification = &svcsdk.StreamSpecification{
+				StreamEnabled: aws.Bool(*r.ko.Spec.StreamSpecification.StreamEnabled),
+			}
+			// Only set streamViewType when streamSpefication is enabled and streamViewType is non-nil.
+			if *r.ko.Spec.StreamSpecification.StreamEnabled && r.ko.Spec.StreamSpecification.StreamViewType != nil {
+				input.StreamSpecification.StreamViewType = aws.String(*r.ko.Spec.StreamSpecification.StreamViewType)
+			}
+		} else {
+			input.StreamSpecification = &svcsdk.StreamSpecification{
+				StreamEnabled: aws.Bool(false),
 			}
 		}
-		removed = append(removed, aElement.Key)
 	}
-	for _, bElement := range b {
-		if !ackutil.InStrings(*bElement.Key, visitedIndexes) {
-			added = append(added, bElement)
+	if delta.DifferentAt("Spec.SSESpecification") {
+		if r.ko.Spec.SSESpecification.Enabled != nil {
```

Comment:
You need a nil guard here on `r.ko.Spec.SSESpecification`. It's causing a nil dereference fault:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x166b8ee]
goroutine 544 [running]:
github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table.(*resourceManager).newUpdateTablePayload(0xc00074def0, {0x1a6373b, 0xc}, 0xc00000e600, 0x0)
	/github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table/hooks.go:299 +0x36e
github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table.(*resourceManager).syncTable(0xc00021d100, {0x1dc6ed8, 0xc00074df20}, 0x0, 0x15bd865)
	/github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table/hooks.go:252 +0x14a
github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table.(*resourceManager).customUpdateTable(0xc00021d100, {0x1dc6ed8, 0xc00074df20}, 0xc00000e600, 0xc00000e6c8, 0xd)
	/github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table/hooks.go:169 +0x565
github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table.(*resourceManager).sdkUpdate(...)
	/github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table/sdk.go:987
github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table.(*resourceManager).Update(0xc00074def0, {0x1dc6ed8, 0xc00074df20}, {0x1df7278, 0xc00000e600}, {0x1df7278, 0xc00000e6c8}, 0xc00108a440)
	/github.com/aws-controllers-k8s/dynamodb-controller/pkg/resource/table/manager.go:148 +0x74
github.com/aws-controllers-k8s/runtime/pkg/runtime.(*resourceReconciler).updateResource(0xc00017ca20, {0x1dc6ed8, 0xc00074df20}, {0x1df3310, 0xc00021d100}, {0x1df7278, 0xc00000e600}, {0x1df7278, 0xc00000e6c8})
	/go/pkg/mod/github.com/aws-controllers-k8s/runtime@v0.17.0/pkg/runtime/reconciler.go:413 +0x305
github.com/aws-controllers-k8s/runtime/pkg/runtime.(*resourceReconciler).Sync(0xc00017ca20, {0x1dc6ed8, 0xc00074df20}, {0x1df3310, 0xc00021d100}, {0x1df7278, 0xc00000e600})
	/go/pkg/mod/github.com/aws-controllers-k8s/runtime@v0.17.0/pkg/runtime/reconciler.go:240 +0x6c5
github.com/aws-controllers-k8s/runtime/pkg/runtime.(*resourceReconciler).reconcile(0xc00017ca20, {0x1dc6ed8, 0xc00074df20}, {0x1df3310, 0xc00021d100}, {0x1df7278, 0xc00000e600})
	/go/pkg/mod/github.com/aws-controllers-k8s/runtime@v0.17.0/pkg/runtime/reconciler.go:191 +0xe5
github.com/aws-controllers-k8s/runtime/pkg/runtime.(*resourceReconciler).Reconcile(0xc00017ca20, {0x1dc6ed8, 0xc000977f20}, {{{0xc00070f6e5, 0x1995580}, {0xc0009e84a0, 0x30}}})
	/go/pkg/mod/github.com/aws-controllers-k8s/runtime@v0.17.0/pkg/runtime/reconciler.go:172 +0x725
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0007e58c0, {0x1dc6ed8, 0xc000977e60}, {{{0xc00070f6e5, 0x1995580}, {0xc0009e84a0, 0x413834}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0007e58c0, {0x1dc6e30, 0xc0005d7dc0}, {0x18af8c0, 0xc000422880})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007e58c0, {0x1dc6e30, 0xc0005d7dc0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x357 
```